dbt/macros/weighted_quantile.sql

```diff
@@ -0,0 +1,26 @@
+{% macro weighted_quantile(model, column_name, weight_col, quantile) %}
```

Comment:
@marianneke this is the macro that's currently broken. It gets called in `expect_column_weighted_quantile_values_to_be_between.sql`. The idea is that this will compute a weighted quantile on a table, then that test will check the quantile vs bounds. I think the issue is with the syntax using `WITH` here. The macro `dbt/macros/check_row_counts_per_partition.sql` has an example where I fixed this to use the proper duckdb syntax.

Reply:
Can you clarify what the syntax issue was and how you solved it in the row count macro?

Reply:
Hmm I'm actually not sure that it was the issue after all. I was getting a similar error message error message with the row count macro and I did have a syntax error there, but this looks ok to me. It might also be something to do with the `expect_column_weighted_quantile_values_to_be_between` macro which is calling this under the hood.